### PR TITLE
8344628: Test TestEnableJVMCIProduct.java run with virtual thread intermittent fails

### PR DESCRIPTION
**Pull request by dougxc**
This PR prevents a rare, intermittent failure of TestEnableJVMCIProduct.java.
It does this by writing the expected test output to a file instead of stdout to avoid issues with VM error logging interleaving with the test output.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344628](https://bugs.openjdk.org/browse/JDK-8344628): Test TestEnableJVMCIProduct.java run with virtual thread intermittent fails (**Bug** - P4)

### Reviewers
 * [SendaoYan](https://openjdk.org/census#syan) (@sendaoYan - Committer)
 * [Dean Long](https://openjdk.org/census#dlong) (@dean-long - **Reviewer**)
 * [Hamlin Li](https://openjdk.org/census#mli) (@Hamlin-Li - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22323/head:pull/22323` \
`$ git checkout pull/22323`

Update a local copy of the PR: \
`$ git checkout pull/22323` \
`$ git pull https://git.openjdk.org/jdk.git pull/22323/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22323`

View PR using the GUI difftool: \
`$ git pr show -t 22323`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22323.diff">https://git.openjdk.org/jdk/pull/22323.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22323#issuecomment-2493837721)
</details>

### CODE REVIEW

**Comment by dougxc** [comment]
@sendaoYan can you please test this fix in your setup.

**Comment by sendaoYan** [comment]
> @sendaoYan can you please test this fix in your setup.

Okey, wait a moment

File: test/hotspot/jtreg/compiler/jvmci/TestEnableJVMCIProduct.java [hunk 1]
```diff
@@ -103,7 +112,11 @@ static void test(String explicitFlag, Expectation... expectations) throws Except
                     output.stdoutShouldMatch("No JVMCI compiler found");
                 }
             } else if (flag.equals("-XX:+UseGraalJIT")) {
-                output.shouldContain("jvmci.Compiler=graal");
```
Comment (dean-long) [inline test/hotspot/jtreg/compiler/jvmci/TestEnableJVMCIProduct.java:106]: Does changing shouldContain() to stdoutShouldContain() also solve the problem?

**Review by sendaoYan — APPROVED** [review]
After apply this patch and run the test with virtual thread 50k times all passed.

File: test/hotspot/jtreg/compiler/jvmci/TestEnableJVMCIProduct.java (hunk 1 above)
Comment (dougxc) [reply test/hotspot/jtreg/compiler/jvmci/TestEnableJVMCIProduct.java:106]: I don't think so as the output is sent to HotSpot's log stream (i.e. `tty`) which I believe goes stdout by default.

**Review by dean-long — APPROVED** [review]

**Review by Hamlin-Li — APPROVED** [review]
Looks good.

**Comment by dougxc** [comment]
Thanks for the reviews.

/integrate

**Pull request closed without merging**